### ISSUE TITLE
Manufacturer Schema does not appear under Tag, Product Categories, Shipping Classes or Product Tag

### ISSUE THREAD

**Issue by Pcosta88** [Yoast/wpseo-woocommerce#128]
**What was expected**
`itemprop="manufacturer"` would appear under all options in the drop-down menu. 

**What happened**
It only appears under that attribute

**How to reproduce**

1.Assign Manufacturer as Attribute from SEO->WooCommerce SEO drop-down menu

![screen shot 2016-07-26 at 12 45 08 pm](https://cloud.githubusercontent.com/assets/8238074/17148055/7d96acee-5333-11e6-9861-7496dc9f3d7a.png)

2.See that `itemprop="manufacturer"` appears

3.Switch Manufacturer to Tag, Product Cat, Shipping Classes Product Tag and see that it does not

**Tech Specs**
WP: 4.5.3
WooComm SEO: 3.4
Yoast SEO Premium: 3.4 (works the same with Yoast SEO)
Themes: Issue remains same on newest 2016 and Storefront

**Comment by boblinthorst** [comment]
Not entirely sure what you're describing here. Could you elaborate? We do not have enough information right now to go on. What's the exact issue and why should it be solved?

**Comment by Pcosta88** [comment]
@boblinthorst Updated with new information.

**Comment by Michelle2403** [comment]
I've tested with release candidate 7.3 and it's still broken.

**Comment by Michelle2403** [comment]
the same issue: https://github.com/Yoast/wpseo-woocommerce/issues/245

**Comment by Pcosta88** [comment]
@galbaras Yes, you can expect this to be resolved. While we are unable to tell you when exactly it will be fixed, you can find the status of this (or any other issue) by viewing the milestone in the top-right. This will let you know the version number the fix will be shipped with.

**Comment by monbauza** [comment]
`itemprop="brand"` isn't outputted either if Brand under SEO > WooCommerce SEO is set to product categories or product tags.

**Comment by galbaras** [comment]
I think this may be an opportunity to change the output from using "itemprop" to adding attributes to the WooCommerce ld+json code using [the "woocommerce_structured_data_context" filter](https://github.com/woocommerce/woocommerce/blob/2c2891c17a54c5daa7cc624c9e1246af1d0e8d6b/includes/class-wc-structured-data.php).

This being a PAID plugin, I find that having an issue open for 2.6 years (!) unacceptable. What are we actually paying for?

**Comment by moorscode** [comment]
It does not really make sense to me, to be able to select "Product Category", "Product Tags", "Shipping classes" or "Tag" for the `manufacturer` connection.
If I'm missing a thought process in this, please let me know.

I lean towards having them removed from the options in this dropdown and only show Product Attributes.
Adding a bit of context on how to add an attribute which can be selected in the settings page would help a lot in the UX of this feature.

It seems to me that adding an attribute to the product that is called "manufacturer" makes a lot more sense and is still manageable.

**Comment by PatrickYoast** [comment]
@moorscode I tend to agree with you as long as we make the UX a lot easier to use and logical for our users. Having said that, I am curious if @jono-alderson shares that opinion or if we are overlooking something.

**Comment by jono-alderson** [comment]
Yep, let's simplify the UX.

We should also move to JSON-LD ASAP, in line with the broader changes we're making elsewhere. Now that WooCommerce have fixed their Product `ID` attribute, there's no reason why our two graphs can't be 'merged' through cross-referencing.

Let's have a whiteboard moment next week?

**Comment by moorscode** [comment]
> Let's have a whiteboard moment next week?

I'll not be here next week, but please do and create a product issue on the tasks needed to implement that holistically :)

**Comment by jono-alderson** [comment]
Ok. Struggling to separate the "Let's revise the UX" and "Let's do all the schema" requirements, so this might end up being quite meaty...

**Comment by moorscode** [comment]
The `itemprop` just works for the current implementation if a Product Attribute is used. This can be implemented separately from changing that it will be placed in the JSON+LD instead of using the `itemprop`.

**Comment by monbauza** [comment]
Related https://github.com/Yoast/wpseo-woocommerce/issues/284

**Comment by rmarcano** [comment]
It seems the "Brand" is not being recognized by the Google Structured Data Tool possibly because of bad formatting : https://github.com/Yoast/wpseo-woocommerce/issues/334

**Comment by galbaras** [comment]
Again, I vote for using filters to set WooCommerce ld+json fields for the output.

For the input, only single values make sense for both Manufacturer and Brand. Since more than one value of Product Categories and Product Tags can be assigned to a product, those are not suitable options. Really, the only options available should be single-value attributes, ideally "Manufacturer" and "Brand".

Check all of the term assignments for every attribute to ensure single-ness is ridiculous, but using only attributes for Manufacturer and Brand seems to make sense, with the possible detection of custom product taxonomies (maybe later?).
